src/basic/terminal-util.c

```diff
@@ -1273,6 +1273,37 @@ int vt_reset_keyboard(int fd) {
         return 0;
 }
 
+int vt_restore(int fd) {
+        static const struct vt_mode mode = {
+                .mode = VT_AUTO,
+        };
+        int r, q = 0;
+
+        r = ioctl(fd, KDSETMODE, KD_TEXT);
+        if (r < 0)
+                q = log_debug_errno(errno, "Failed to set VT in text mode, ignoring: %m");
+
+        r = vt_reset_keyboard(fd);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to reset keyboard mode, ignoring: %m");
+                q = (q != 0) ? q : r;
```

Comment:
not that it matters, but not need for those `()`. also i think it would be slightly nicer to write this as:

```c
if (q >= 0)
        q = r;
```

Reply:
I put them to improve the readability but it just looks better to use an `if()` block as you suggested.